### ISSUE TITLE
AutoIncrement of Primary ID not working

### ISSUE THREAD

**Issue by Snogo** [capacitor-community/sqlite#84]
somehow the AutoIncrement for any primary ID is not working for me. I tried two different tables and different times to INSERT a new row with different values. also when inserting two rows subsequently with the same values. the `ret.changes.lastId` is always **1**. I also tried to include the word AUTOINCREMENT in my create table sql like `id INTEGER PRIMARY KEY _AUTOINCREMENT_ NOT NULL` with no effect

I know you that you exactly test that in your examples (e. g. _NoEncryptionTest.vue_). however with my examples like here it is not working:

```
let values: Array<any> = [sampleId.value, animalNr, lat, long, currentDateTime ];
ret = await db.run(createNewOccurence,values);
let lastOccurenceId = ret.changes.lastId; //new id 1
console.log("new oc id1: " + lastOccurenceId);
```

```
values = [sampleId.value, animalNr, lat, long, currentDateTime ];
ret = await db.run(createNewOccurence,values);
lastOccurenceId = ret.changes.lastId; //new id 2
console.log("new oc id2: " + lastOccurenceId);
```
the Insert works. the _runSQL Changes is 1_. the result however in `lastOccurenceId ` is always 1. I don't know how to give you more details
___
Windows10. "@capacitor-community/sqlite": "^2.9.12". "vue-sqlite-hook": "^1.0.1". "@ionic/vue": "^5.5.4"
Samsung Galax S7. Android

**Comment by jepiqueau** [comment]
@Snogo this is strange , can you share the table schema and the createNewOccurrence string so i can try to reproduce it?

**Comment by Snogo** [comment]
```
CREATE TABLE IF NOT EXISTS occurence (
                id INTEGER PRIMARY KEY AUTOINCREMENT NOT NULL, 
                sample_nr INTEGER,
                animal_nr INTEGER,
                count INTEGER,
                latitude FLOAT,
                longitude FLOAT,
                datetime DATETIME
                );
```

```
export const createNewOccurence = `
    INSERT INTO occurence (sample_nr, animal_nr, count, latitude, longitude, datetime) VALUES (?,?,1,?,?,?);
`;
```

**Comment by jepiqueau** [comment]
@Snogo look at `https://www.sqlite.org/datatype3.html` you will see that the DATETIME is not a SQLite types.
```
2.2. Date and Time Datatype
SQLite does not have a storage class set aside for storing dates and/or times. Instead, the built-in Date And Time Functions of SQLite are capable of storing dates and times as TEXT, REAL, or INTEGER values:

TEXT as ISO8601 strings ("YYYY-MM-DD HH:MM:SS.SSS").
REAL as Julian day numbers, the number of days since noon in Greenwich on November 24, 4714 B.C. according to the proleptic Gregorian calendar.
INTEGER as Unix Time, the number of seconds since 1970-01-01 00:00:00 UTC.
Applications can chose to store dates and times in any of these formats and freely convert between formats using the built-in date and time functions.
```
it should be the reason. me i used an Integer where i stored the unix epochs date

**Comment by Snogo** [comment]
thank you for pointing that out. you are right. I changed it to **TEXT** and use the format "DD.MM.YYYY HH:MM:SS" and I am more conscious about using dates in general but still the `lastId ` is still **1** unfortunately. also in another table:

```
CREATE TABLE IF NOT EXISTS sample (
            id INTEGER PRIMARY KEY AUTOINCREMENT NOT NULL, 
            user_nr INTEGER,
            date_begin TEXT,
            date_end TEXT,
            notes TEXT
            );
```
do you have another idea?

**Comment by jepiqueau** [comment]
@Snogo this has been fixed in 2.9.13 see `https://github.com/jepiqueau/vue-sqlite-app-starter/tree/refactor`for validation test

**Comment by Snogo** [comment]
@jepiqueau it works! thank you very much for the support and fix. I will close this issue
